pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -31,265 +30,183 @@ import (
 	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
 )
 
-type AgentUpdateEvent struct {
-	InterfaceStatuses *[]api.InterfaceStatus
-	DomainName        string
-	OsInfo            *api.GuestOSInfo
-}
+// AgentCommand is a command exutable on guest agent
+type AgentCommand string
 
-type AgentPoller struct {
-	Connection      cli.Connection
-	VmiUID          types.UID
-	domainData      *DomainData
-	agentDone       chan struct{}
-	domainUpdate    chan *api.Domain
-	pollTime        time.Duration
-	agentUpdateChan chan AgentUpdateEvent
-}
+// Aliases for commands executed on guest agent
+// when updated to libvirt 5.6.0 this can change to libvirt types
+// Aliases are also used as keys to the store, it does not matter how the keys are named,
+// only whther it relates to the right data
+const (
+	GET_OSINFO     AgentCommand = "guest-get-osinfo"
+	GET_INTERFACES AgentCommand = "guest-network-get-interfaces"
+)
 
-type DomainData struct {
-	name       string
-	osInfo     api.GuestOSInfo
-	aliasByMac map[string]string
-	interfaces []api.InterfaceStatus
+// AgentUpdatedEvent fire up when data is changes in the store
+type AgentUpdatedEvent struct {
+	Type       AgentCommand
+	DomainInfo api.DomainGuestInfo
 }
 
-// Result for json unmarshalling
-type Result struct {
-	Interfaces []Interface `json:"return"`
+// AgentStore stores the agent data converted to api domain objects
+// it offers methods to get the data and fire up an event when there
+// is a change of the data
+type AgentStore struct {
+	store        sync.Map
+	AgentUpdated chan AgentUpdatedEvent
 }
 
-// Guest OS Info unmarshalling
-type ParsedGuestOsInfo struct {
-	Name          string `json:"name"`
-	KernelRelease string `json:"kernel-release"`
-	Version       string `json:"version"`
-	PrettyName    string `json:"pretty-name"`
-	VersionId     string `json:"version-id"`
-	KernelVersion string `json:"kernel-version"`
-	Machine       string `json:"machine"`
-	Id            string `json:"id"`
+// NewAgentAstore creates new agent store
+func NewAgentAstore() AgentStore {
```

Comment:
Why is it `AgentAstore`? My guess: you mean `AsyncAgentStore`? Then either you should write `NewAgentAStore`, but `NewAsyncAgentStore` would be more clear.